src/app/app.component.ts

```diff
@@ -1,11 +1,33 @@
-import { Component} from '@angular/core';
+import { Component, OnInit} from '@angular/core';
+import { Router, NavigationEnd, ActivatedRoute } from '@angular/router';
+import { Title } from '@angular/platform-browser';
+import { filter, map, mergeMap} from 'rxjs/operators';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
-  constructor() {
+export class AppComponent implements OnInit {
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private titleService: Title
+  ) {}
+
+  ngOnInit() {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      map(() => this.activatedRoute),
+      map((route) => {
+        while (route.firstChild) {
+          route = route.firstChild;
+        }
+        return route;
+      }),
+      mergeMap((route) => route.data),
+    ).subscribe((event) => this.titleService.setTitle(event['title']));
```

Comment:
I know this is the appComponent but unsubscribe anyways...just in case

Reply:
Added unsubscribe in latest commit.